readthedocs/builds/signals.py

```diff
@@ -2,7 +2,28 @@
 
 """Build signals."""
 
-import django.dispatch
+import logging
+import os
 
+from django.conf import settings
+from django.dispatch import receiver, Signal
+from django.utils.autoreload import file_changed
 
-build_complete = django.dispatch.Signal(providing_args=['build'])
+from readthedocs.worker import app
+
+build_complete = Signal(providing_args=['build'])
+
+
+log = logging.getLogger(__file__)
+
+
+if all([
+        # restart workers in local docker compose only
+        settings.RTD_DOCKER_COMPOSE,
+        # restart workers only from web instance
+        os.environ['DJANGO_SETTINGS_MODULE'].endswith('web_docker'),
```

Comment:
Why are we only restarting the webs?

Reply:
We are restarting all the workers.

This check is here because proxito and web will be checking for file changes in disk and this signal will be triggered from both, and we want to restart the workers only once. I just picked `web` but I could have picked `proxito`.